src/Microsoft.Management.Deployment/PackageCatalogReference.cpp

```diff
@@ -124,7 +130,17 @@ namespace winrt::Microsoft::Management::Deployment::implementation
         }
         return GetConnectCatalogErrorResult();
     }
+    winrt::Windows::Foundation::Collections::IVectorView<winrt::Microsoft::Management::Deployment::SourceAgreement> PackageCatalogReference::SourceAgreements()
+    {
+        for (auto const& agreement : m_sourceReference.GetInformation().SourceAgreements)

```

Comment:
This logic should be guarded by once_flag, otherwise if caller calls this api multiple times, the m_sourceagreements keeps adding duplicate content, and it's not thread safe.

Reply:
added once-flag to guard this logic.